src/backend/mod.rs

```diff
@@ -34,7 +34,9 @@ pub trait Backend {
     fn clear(&self, color: theme::Color);
 
     fn set_refresh_rate(&mut self, fps: u32);
-    // TODO: unify those into a single method?
-    fn with_color<F: FnOnce()>(&self, colors: theme::ColorPair, f: F);
-    fn with_effect<F: FnOnce()>(&self, effect: theme::Effect, f: F);
+
+    fn set_color(&self, colors: theme::ColorPair) -> theme::ColorPair;
```

Comment:
Can we add a comment here to indicate that the backend should return the previous state?